torch/ao/quantization/fx/prepare.py

```diff
@@ -244,18 +265,14 @@ def is_pattern_dtype_config_supported_by_backend(
     for dtype_config in dtype_configs:
         # check if arg dtype are supported
         supported = True
-        for arg in input_node.args:
-            supported = supported and \
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
-        for k, arg in input_node.kwargs.items():
-            supported = supported and \
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
+        for arg in list(input_node.args) + list(input_node.kwargs.values()):
+            supported = supported and is_input_arg_dtype_supported_by_backend(
+                arg, input_node, node_name_to_target_dtype, activation, dtype_config, backend_config)
         # check if output dtype is supported
         supported = supported and is_output_dtype_supported_by_backend(
-            output_node, node_name_to_target_dtype, dtype_config)
+            output_node, node_name_to_target_dtype, activation, dtype_config)
         if supported:
+            _maybe_validate_activation_post_process_dtype_constraints(weight, dtype_config.weight_dtype)
```

Comment:
didn't we check weight dtype before?

Reply:
looks like we don't... I'll add an issue for this one

Reply:
ideally this should be merged into `is_input_arg_dtype_supported_by_backend` I think, since that handles both input activation and weight previously